### ISSUE TITLE
‘This sign-in method is not enabled’ when signing in with an email or mobile phone number.

### ISSUE THREAD

**Issue by KnightJun** [logto-io/logto#6347]
### Describe the bug
I've been using the ‘Social Login’ method for my application, and my registration ID is set to ‘None’, now I want to add a mobile phone number/email as a new login method, and I found that after adding it, I will be prompted when logging in a new mobile phone number: ‘This sign-in method is not enabled’. Only when I set the sign-in method to email address or mobile phone number, I can register properly, but after setting the sign-in method, users who have previously registered using the social method will need to register their email or mobile phone when logging in. I think this is a bad experience and old users will think I'm collecting their information! It also causes inconvenience to some regular users who do not want to provide their mobile phone numbers
我的程序之前一直是使用“社交登录”方式，我的注册标识设置的是“无”，现在我想增加手机号/邮箱作为新的登录方式，发现增加后在登录新的手机号时会提示：“This sign-in method is not enabled”。只有当我把注册标识设置为邮件地址或手机号码才能正常注册，但设置了注册标识后，之前使用社交方式注册登录的用户在登录时需要登记他们的电子邮箱或者手机。我认为这是很不好的体验，老用户会认为我在收集他们的信息，同时也对部分不想提供手机号码的老用户造成了不便

### Expected behavior
I would like to be able to register and login with a mobile phone number if the registration mark is set to ‘none’.
希望在注册标识设置为“无“的情况下能正常使用手机号码注册登录

### How to reproduce?
If you configure the mobile phone number login method when the registration mark is set to ‘None’, you cannot login properly when you register with a new mobile phone number.
注册标识设置为“无“的情况下，配置手机号码登录方式，使用新的手机号码注册登录会无法正常登录

### Context
Logto 1.1.0

- [ ] Logto Cloud
- [ √] Self-hosted, Logto version = 
  - [√ ] Container (Docker image)
  - [ ] Raw Node.js

### Screenshots
![image](https://github.com/user-attachments/assets/678a6d70-dda4-46c0-9e59-571177e86915)
![image](https://github.com/user-attachments/assets/b15940d5-047e-4dc3-8a4f-d97cd1d6bc3a)

**Comment by xiaoyijun** [comment]
Hi @KnightJun , thanks for your feedback, I'll discuss this with the team.

**Comment by xiaoyijun** [comment]
Hi @KnightJun , thanks for your feedback.
We have confirmed that prompting the user to create an account when the sign-up method is not supported is a bug, and it has already been fixed in the codebase. This fix will be released in the next version.

As for your request to have existing social users provide their phone numbers, we will launch a new project to enhance the flexibility of the Sign-in experience configuration. cc @fleuraly @Rany0101
